backend/src/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -0,0 +1,81 @@
+package org.commcare.session;
+
+import org.commcare.suite.model.DisplayUnit;
+import org.commcare.suite.model.RemoteQueryDatum;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.xpath.expr.XPathExpression;
+import org.javarosa.xpath.expr.XPathFuncExpr;
+
+import java.util.Enumeration;
+import java.util.Hashtable;
+
+/**
+ * Manager for remote query datums; get/answer user prompts and build
+ * resulting query url.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class RemoteQuerySessionManager {
+    private final RemoteQueryDatum queryDatum;
+    private final Hashtable<String, String> userAnswers = new Hashtable<String, String>();
+    private final EvaluationContext evaluationContext;
+
+    public RemoteQuerySessionManager(RemoteQueryDatum queryDatum, EvaluationContext evaluationContext) {
+        this.queryDatum = queryDatum;
+        this.evaluationContext = evaluationContext;
+    }
+
+    public Hashtable<String, DisplayUnit> getNeededUserInputDisplays() {
+        return queryDatum.getUserQueryPrompts();
+    }
+
+    public Hashtable<String, String> getUserAnswers() {
+        return userAnswers;
```

Comment:
Is it possible for the user answers to be pre-populated in the same manner as the query params, but in a way that can be overriden, or is it a binary thing (some params are functions, some are user input)?